src/module-elasticsuite-core/Healthcheck/ReplicasConfigCheck.php

```diff
@@ -0,0 +1,167 @@
+     */
+    private const ES_INDICES_SETTINGS_WIKI_PAGE = 'https://github.com/Smile-SA/elasticsuite/wiki/ModuleInstall#indices-settings';
+
+    public const WARNING_STATUS = 'warning';
+
+    /**
+     * @var IndexSettingsHelper
+     */
+    protected $helper;
+
+    /**
+     * @var Client
+     */
+    protected $client;
+
+    /**
+     * @var UrlInterface
+     */
+    private $urlBuilder;
+
+    /**
+     * Constructor.
+     *
+     * @param IndexSettingsHelper $indexSettingHelper Index settings helper.
+     * @param Client              $client             Elasticsearch client.
+     * @param UrlInterface        $urlBuilder         URL builder.
+     */
+    public function __construct(
+        IndexSettingsHelper $indexSettingHelper,
+        Client $client,
+        UrlInterface $urlBuilder
+    ) {
+        $this->helper = $indexSettingHelper;
+        $this->client = $client;
+        $this->urlBuilder = $urlBuilder;
+    }
+
+    /**
+     * Retrieve the unique identifier for this health check.
+     *
+     * @return string
+     */
+    public function getIdentifier(): string
+    {
+        return 'replicas_config_check';
+    }
+
+    /**
+     * Retrieve a dynamic description for this health check based on its status.
+     *
+     * @return string
+     */
+    public function getDescription(): string
+    {
+        $status = $this->getStatus();
+
+        if ($status === self::WARNING_STATUS) {
+            // Description when the replicas configuration is incorrect.
+            // @codingStandardsIgnoreStart
+            return __(
+                'The number of replicas configured for Elasticsuite <strong>is incorrect.</strong> '
+                . 'You can\'t use <strong>%1 replicas</strong> since there is only <strong>%2 nodes</strong> in your Elasticsearch cluster.<br/>'
+                . 'Click <a href="%3"><strong>here</strong></a> to go to the <strong>Elasticsuite Basic Settings</strong> page and change your <strong>Number of Replicas per Index</strong> parameter according to our <a href="%4" target="_blank"><strong>Wiki page</strong></a>.',
+                $this->helper->getNumberOfReplicas(),
+                $this->getNumberOfNodes(),
+                $this->getElasticsuiteConfigUrl(),
+                self::ES_INDICES_SETTINGS_WIKI_PAGE
+            );
+            // @codingStandardsIgnoreEnd
+        }
+
+        // Description when the replicas configuration is optimized.
+        return __('The number of replicas is properly configured for the Elasticsearch cluster. No action is required at this time.');
+    }
+
+    /**
+     * Retrieve the status of this health check.
+     *
+     * @return string
+     */
+    public function getStatus(): string
+    {
+        $numberOfReplicas = $this->helper->getNumberOfReplicas();
+        $numberOfNodes = $this->getNumberOfNodes();
+
+        if ($numberOfReplicas > 1 && $numberOfReplicas > ($numberOfNodes - 1)) {
+            return 'warning';
```

Comment:
Legacy check was actually wrong.
"$numberOfReplicas > ($numberOfNodes - 1)" is ok.
But the first part should be "$numberOfReplicas > 0".
=> "If the number of replicas > 0 (it means we enabled replicas) then, there is a warning if there are more replicas than available nodes".